player/tanks.py

```diff
@@ -0,0 +1,108 @@
+from typing import *
+from client.responses import *
+from player.player import *
+from ai.pathFinder import AStarPathfinding
+from model.hex import Hex as MapHex
+
+
+def check_range(game_actions: GameActionsResponse, my_vehicle: Vehicle, enemy_vehicle: Vehicle) -> bool:
+    if my_vehicle.vehicle_type == VehicleType.MEDIUM_TANK:
+        return Medium.is_enemy_in_range(game_actions, my_vehicle, enemy_vehicle)
+    elif my_vehicle.vehicle_type == VehicleType.HEAVY_TANK:
+        return Heavy.is_enemy_in_range(game_actions,my_vehicle, enemy_vehicle)
+    elif my_vehicle.vehicle_type == VehicleType.LIGHT_TANK:
+        return Light.is_enemy_in_range(game_actions,my_vehicle, enemy_vehicle)
+    elif my_vehicle.vehicle_type == VehicleType.SPG:
+        return SPG.is_enemy_in_range(game_actions,my_vehicle, enemy_vehicle)
+    elif my_vehicle.vehicle_type == VehicleType.AT_SPG:
+        return AT_SPG.is_enemy_in_range(game_actions,my_vehicle, enemy_vehicle)
+    return False
+
+
+class Medium:
+    @staticmethod
+    def is_enemy_in_range(game_actions: GameActionsResponse, my_vehicle: Vehicle, enemy_vehicle: Vehicle):
```

Comment:
This method seems to be a mandatory part of a public interface for all objects of similar classes. But how do you enforce this method to be implemented in all classes? Maybe all those classes shall inherit from ABC. Or you may consider using Protocol from typing.